src/gui/uithememanager.cpp

```diff
@@ -32,13 +32,70 @@
 #include <QApplication>
 #include <QFile>
 #include <QIcon>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QPalette>
 #include <QResource>
 
 #include "base/iconprovider.h"
 #include "base/logger.h"
 #include "base/preferences.h"
 #include "base/utils/fs.h"
 
+namespace
+{
+    void loadPaletteFromUITheme(QPalette &dest
+                                , const std::initializer_list<std::pair<QString, QPalette::ColorRole>> &keyAndColorRoles
+                                , const QPalette::ColorGroup colorGroup)
+    {
+        for (const std::pair<QString, QPalette::ColorRole> &keyColorRolePair : keyAndColorRoles) {
+            const QColor defaultColor = dest.color(colorGroup, keyColorRolePair.second);
+            const QColor newColor = UIThemeManager::instance()->getColor(keyColorRolePair.first, defaultColor);
+            dest.setColor(keyColorRolePair.second, newColor);
+        }
+    }
+
+    void loadCustomPalette()
+    {
+        const std::initializer_list<std::pair<QString, QPalette::ColorRole>> keyAndColorRolesNormalGroup =
```

Comment:
I don't think `std::initializer_list` is supposed to be used this way.

Reply:
which ds should I use then? QVector will dynamically allocate, may be std::array? but then will have to templatized `loadPaletteFromUITheme`

Reply:
or may be `std::pair<QString, QPalette::ColorRole>[]`

Reply:
>or may be std::pair<QString, QPalette::ColorRole>[]

Or `QPair<QString, QPalette::ColorRole>[]`, weird to see mixing std and Qt containers.
IMO current C++14 is not-so-easy to get constexpr done, better wait for C++17 or even C++20, so don't give yourself a hard time here.